### PR TITLE
Formalize flake8 linting

### LINKED ISSUE #1435: Set up flake8 code check properly
Move flake8 configuration to file and do not exit with 0 code in CI.

### PR DESCRIPTION
**Pull request by mondeja**
Closes #1435

+ Added `lint` optional requirements to `setup.py`.
+ Fixed all linting problems in `tests/`, `moviepy/` and `examples/`.
+ Relative imports converted to absolute ones (this is linted also).
+ Added documentation about how to format and lint the code in README.
+ Moved CI flake8 configuration to `setup.cfg` file.
+ Separated CI linting in 4 different steps:
  - `scripts` (`setup.py`, `docs/conf.py` and `find_latest_imagemagick_version.py`)
  - `examples`
  - `tests`
  - `moviepy`
+ Removed uneeded `# -*- coding: utf-8 -*-` statements.
+ Removed uneeded `__future__` imports.

- [x] I have properly explained unusual or unexpected code in the comments around it
- [x] I have formatted my code using `black -t py36`

### CODE REVIEW

**Review by tburrows13 — APPROVED** [review]

**Comment by mondeja** [comment]
Blocked until #1453 merges.

**Pull request merged**
